libraries/AP_GPS/examples/GPS_AUTO_test/GPS_AUTO_test.cpp

```diff
@@ -16,42 +16,42 @@
      Test for AP_GPS_AUTO
 */
 
-#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/AP_HAL.h>                                      //This is a common Hardware Abstraction Layer.
 #include <AP_GPS/AP_GPS.h>
 #include <GCS_MAVLink/GCS_Dummy.h>
 #include <AP_Notify/AP_Notify.h>
 #include <AP_Notify/AP_BoardLED.h>
 #include <AP_SerialManager/AP_SerialManager.h>
 #include <AP_BoardConfig/AP_BoardConfig.h>
 
-void setup();
-void loop();
+void setup();                                                   //This function is defined in most of the libraries. This function is called only once at boot up time. This function is called by main() function in HAL.
+void loop();                                                    //This function is defined in most of the libraries. This function is called by main function in HAL. The main work of the sketch is typically in this function only.
 
-const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();                     //Declare "hal" reference variable. This variable is pointing to Ap_HAL::HAL class's object. Here, AP_HAL is library and HAL is a class in that library. This reference variable can be used to get access to hardware specific functions.                     
 
-static AP_BoardConfig board_config;
+static AP_BoardConfig board_config;                             //Declare a static object for configuring a board.
 
 // create board led object
-AP_BoardLED board_led;
+AP_BoardLED board_led;                                          //Declare a BoardLED variable.
 
 // create fake gcs object
-GCS_Dummy _gcs;
+GCS_Dummy _gcs;                                                 //gcs stands for Ground Control Station
 
 const AP_Param::GroupInfo GCS_MAVLINK_Parameters::var_info[] = {
         AP_GROUPEND
 };
 
 // This example uses GPS system. Create it.
-static AP_GPS gps;
+static AP_GPS gps;                                              //Declare a GPS variable.
 // Serial manager is needed for UART communications
 static AP_SerialManager serial_manager;
 
 //to be called only once on boot for initializing objects
 void setup()
 {
-    hal.console->printf("GPS AUTO library test\n");
+    hal.console->printf("GPS AUTO library test\n");             //To indicate that GPS AUTO library is being used.
 
-    board_config.init();
+    board_config.init();                                        //call an init() function to initialize a board.
```

Comment:
Doesn't help as a comment.

Comments should NOT just repeat what the code does in English unless it's *really* rather unclear what's going on.  Comments should generally be the "why" rather than that "what".